adapters/boldwin_rapid/boldwin_rapid.go

```diff
@@ -0,0 +1,172 @@
+package boldwin_rapid
+
+import (
+	"encoding/json"
+	"fmt"
+	"text/template"
+
+	"net/http"
+
+	"github.com/prebid/openrtb/v20/openrtb2"
+	"github.com/prebid/prebid-server/v3/adapters"
+	"github.com/prebid/prebid-server/v3/config"
+	"github.com/prebid/prebid-server/v3/macros"
+	"github.com/prebid/prebid-server/v3/openrtb_ext"
+	"github.com/prebid/prebid-server/v3/util/jsonutil"
+)
+
+type adapter struct {
+	//endpoint string
+	endpoint *template.Template
+}
+
+func Builder(_ openrtb_ext.BidderName, config config.Adapter, _ config.Server) (adapters.Bidder, error) {
+	endpointTemplate, err := template.New("endpointTemplate").Parse(config.Endpoint)
+	if err != nil {
+		return nil, fmt.Errorf("unable to parse endpoint template: %v", err)
+	}
+
+	bidder := &adapter{
+		endpoint: endpointTemplate,
+	}
+
+	return bidder, nil
+}
+
+func (a adapter) buildEndpointURL(boldwinExt openrtb_ext.ImpExtBoldwinRapid) (string, error) {
+	endpointParams := macros.EndpointTemplateParams{
+		PublisherID: boldwinExt.Pid,
+		PlacementID: boldwinExt.Tid,
+	}
+
+	return macros.ResolveMacros(a.endpoint, endpointParams)
+}
+
+func (a *adapter) MakeRequests(request *openrtb2.BidRequest, _ *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	var adapterRequests []*adapters.RequestData
+
+	reqCopy := *request
+
+	for _, imp := range request.Imp {
+		// Create a new request with just this impression
+		reqCopy.Imp = []openrtb2.Imp{imp}
+
+		var bidderExt adapters.ExtImpBidder
+		var boldwinExt openrtb_ext.ImpExtBoldwinRapid
+
+		// Use the current impression's Ext
+		if err := jsonutil.Unmarshal(imp.Ext, &bidderExt); err != nil {
+			return nil, []error{err}
+		}
+
+		if err := jsonutil.Unmarshal(bidderExt.Bidder, &boldwinExt); err != nil {
+			return nil, []error{err}
+		}
+
+		endpoint, err := a.buildEndpointURL(boldwinExt)
+		if err != nil {
+			return nil, []error{err}
+		}
+
+		adapterReq, err := a.makeRequest(&reqCopy, endpoint)
+		if err != nil {
+			return nil, []error{err}
+		}
+
+		if adapterReq != nil {
+			adapterRequests = append(adapterRequests, adapterReq)
+		}
+	}
+
+	return adapterRequests, nil
+}
+
+func (a *adapter) getHeaders(request *openrtb2.BidRequest) http.Header {
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+	headers.Add("Accept", "application/json")
+	headers.Add("x-openrtb-version", "2.5")
+	headers.Add("Host", "rtb.beardfleet.com") // required header for the request
```

Comment:
Why is the address hard coded? What if the endpoint is overwritten? Should this value be pulled from the endpoint url configured for this adapter?